packages/core/admin/admin/src/content-manager/components/Wysiwyg/MediaLibrary.js

```diff
@@ -1,58 +1,37 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import {
-  ModalLayout,
-  ModalHeader,
-  ModalBody,
-  ModalFooter,
-} from '@strapi/design-system/ModalLayout';
-import { Button } from '@strapi/design-system/Button';
-import { Text } from '@strapi/design-system/Text';
+import { useLibrary } from '@strapi/helper-plugin';
 
-// const image = [
-//   { alt: 'sunrise', url: 'http://localhost:3000/sunriseimage' },
-//   { alt: 'sunset', url: 'http://localhost:3000/sunsetimage' },
-// ];
+const Steps = {
+  SelectAsset: 'SelectAsset',
+  UploadAsset: 'UploadAsset',
+};
+
+const MediaLibrary = ({ onClose, onSelectAssets }) => {
```

Comment:
I think this abstraction should be done in the upload plugin.

The upload plugin when registering to the admin panel should add the `MediaLibrary` component:

```js
addComponent({ name: 'medial-library', Component: MediaLibrary });
```


Then the only props needed to make the MediaLibrary in the CM would be the `onValidate` and `onClose` `allowedType`.

Reply:
This has one cons: people have to override the whole component (both steps) instead of one. Is this something we want?